### DISCUSSION TITLE
What does JAX stand for?

### DISCUSSION THREAD

**Discussion by surak** [jax-ml/jax#9019] [General] [answered]
The acronym is not clear, and not mentioned anywhere. I suspect that A is from autograd, and X is from XLA, but the J?

**Comment by grahamgower** [comment]
Jit?

**Comment by surak** [comment]
I thought so, but JIT is already covered by the "X" from "XLA"

**Comment by hawkinsp** [comment, marked as answer]
@mattjj @froystig 

Originally, it stood for "Just After eXecution", which refers to how JAX used to do tracing of functions. But JAX doesn't do it that way any more!

A more recent reinterpretation of JAX as an acronym is "JAX is Autograd and XLA". Even that's a bit inaccurate: JAX isn't the [autograd project](https://github.com/HIPS/autograd) itself, instead its spiritual successor. JAX can also use both XLA and other compilers, so even the XLA part is a bit misleading.

So I think we don't really use JAX as an acronym anymore. JAX is JAX.

**Reply by mattjj** [reply]
I agree! Though actually I think "JAX is Autograd and XLA" is not so recent. In fact, JAX _used_ to be built directly on top of [https://github.com/hips/autograd](Autograd), when it was just a baby project in 2017, though that hasn't been the case for a long time. Also, even though it's not literally accurate, that acronym is enshrined in the first line of the README:

![image](https://user-images.githubusercontent.com/1458824/146809783-f4d1dcd5-d800-43af-9655-132aca74c81e.png)

I think Peter's right that there is no official acronym. Except we can retcon whatever acronym seems useful or humorous in the moment!

**Reply by jakevdp** [reply]
It stands for "JAX is an Acronym? ❌ "

**Reply by niklasschmitz** [reply]
I was pretty sure it was "Jacs" for Jacobians and xla!

**Reply by kasra-keshavarz** [reply]
"JAX Ain't XLA" Random post from the internet, didn't mean to be spam

**Reply by jpienaar** [reply]
Funnily enough I see folks have now started expanding it to Just Another XLA.

**Comment by NathanWycoff** [comment]
If there's no official acronym, can I humbly submit "Just Another XLA"?

This is in the spirit of "JAGS" - Just Another Gibbs Sampler, which is really anything but.

**Reply by mattjj** [reply]
Ah yes that was one we had too! "Just Another XLA-frontend".
